stacks-signer/src/client/stackerdb.rs

```diff
@@ -90,8 +115,146 @@ impl StackerDB {
         }
     }
 
+    /// Get the latest signer transactions from signer ids
+    pub fn get_signer_transactions_with_retry(
+        &mut self,
+        signer_ids: &[u32],
+    ) -> Result<Vec<StacksTransaction>, ClientError> {
+        let slot_ids: Vec<_> = signer_ids
+            .iter()
+            .map(|id| id * SIGNER_SLOTS_PER_USER + TRANSACTIONS_SLOT_ID)
+            .collect();
+        debug!(
+            "Getting latest chunks from stackerdb for the following signers: {:?}",
+            signer_ids
+        );
+        let send_request = || {
+            self.signers_stackerdb_session
+                .get_latest_chunks(&slot_ids)
+                .map_err(backoff::Error::transient)
+        };
+        let chunk_ack = retry_with_exponential_backoff(send_request)?;
+        let mut transactions = Vec::new();
+        for (i, chunk) in chunk_ack.iter().enumerate() {
+            if let Some(data) = chunk {
+                if let Ok(message) = bincode::deserialize::<SignerMessage>(data) {
+                    if let SignerMessage::Transactions(chunk_transactions) = message {
```

Comment:
Kind of a nit, but I think it's better to try to avoid deep nesting like this by inverting the checks or using `let...else`:

```rust
let Some(data) = chunk else {
    continue;
}
let Ok(message) = bincode::deserialize::<SignerMessage>(data) else {
    warn!("Failed to deserialize chunk data into a SignerMessage");
    continue;
}
...
```